### ISSUE TITLE
Color Difference

### ISSUE THREAD

**Issue by jclgs** [mozilla/pdf.js#8500]
Configuration:
- Chrome 58, Firefox 53, IE 11
- Windows 7
- PDF.js version: 1.4.20

Steps to reproduce the problem:
1. Open a PDF with Color in Adobe Reader
2. Open the same PDF with PDF.js

What is the expected behavior? 
Color in PDF.js should be the same as Adobe Reader when both viewer are open on the same monitor.

What went wrong? (add screenshot)
Color in PDF.js do not match color in Adobe Reader, on the same Monitor.

![image](https://user-images.githubusercontent.com/24614035/26889423-23e55258-4b7c-11e7-9ae5-463207050115.png)

**Comment by yurydelendik** [comment]
The color might differ a little for CMYK colors and/or overprint. We need to look at the sample PDF to certainly tell the reason.

**Comment by jclgs** [comment]
Is there a way for me that I can transfer you securely the PDF?

**Comment by yurydelendik** [comment]
Prepare sample non-secret pdf, so any contributor can look at it. (There is no value for the project for only me to have such pdf)

**Comment by jclgs** [comment]
[ColorIssue.pdf](https://github.com/mozilla/pdf.js/files/1058844/ColorIssue.pdf)

**Comment by jclgs** [comment]
Is there any development on that case? Is this something that can be solved?
Thanks

**Comment by brendandahl** [comment]
I took a quick glance at the document structure and it appears to use ICC color profiles, which pdf.js does support so we fall back to using CMYK colorspace conversion. CMYK conversion also isn't perfect (see https://github.com/mozilla/pdf.js/issues/7905#issuecomment-281753332).

**Comment by photopea** [comment]
Is there any chance that pdf.js will start to "understand" ICC color profiles attached to images? Instead of expecting everything to be in sRGB.

I am working on my own ICC library (in Javascript), would you be interested in using it for pdf.js?

**Comment by yurydelendik** [comment]
> I am working on my own ICC library (in Javascript), would you be interested in using it for pdf.js?

It depends how you will license your code, but it will be nice to support ICC color profiles.

**Comment by photopea** [comment]
I publish all such code under MIT licence. So I will let you know once it is ready.

**Comment by brianholle** [comment]
@photopea Has there been any progress on this / a solution to the above issue? 

-Thanks

**Comment by SamyCookie** [comment]
duplicate of #2856 ?

**Comment by photopea** [comment]
@brianholle I was stuck on some problem for 3 days and did not find any ICC expert who could help me, so I gave it up for that time. I tried to contatc authors of ICC libraries and authors of ICC standard, but nobody replied.

**Comment by timvandermeij** [comment]
Closing since this is a duplicate of #2856, which tracks support for ICC.
